### PR TITLE
[DB-12356] Bug fix: Name registry initialisation errors out on pg_sequences for PostgreSQL 9

### PR DESCRIPTION
**Pull request by priyanshi-yb**
**Description**: 

`pg_sequences` table is introduced in PG10 which includes the information of normal CREATE SEQUENCE and the sequence by identity columns (these type of columns are introduced in PG10 itself), so for PG9 we can use `information_schema.sequences` and query for the list of sequences.

Fixes - https://yugabyte.atlassian.net/browse/DB-12356

### CODE REVIEW

**Review by makalaaneesh — APPROVED** [review]
LGTM! Did you try running our github action tests with PG 9?

**Comment by priyanshi-yb** [comment]
@makalaaneesh, yes I did try to run the tests but there were issues with most of the test cases, so just got the datatypes and constraints working so thought it should be fine.

**Comment by makalaaneesh** [comment]
> @makalaaneesh, yes I did try to run the tests but there were issues with the schema of most of the test cases, so just got the datatypes and constraints working so thought it should be fine.

Got it. Can you add automation for that? i.e. only like 1/2 tests for PG 9 (offline, live both) in github actions, so that we know that voyager works with PG 9, and can catch any regressions in future.

File: .github/workflows/pg-9-migtests.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: "PG: Migration Tests"
```
Comment (sanyamsinghal) [inline .github/workflows/pg-9-migtests.yml:1]: if we are going to have two workflows for PG.
Can we name it like:
PG[9]: Migration Tests
PG[13]: Migration Tests

File: yb-voyager/src/srcdb/postgres.go [hunk 2]
```diff
@@ -432,10 +432,20 @@ func (pg *PostgreSQL) GetAllSequences() []string {
...
 func (pg *PostgreSQL) GetAllSequencesRaw(schemaName string) ([]string, error) {
 	var sequenceNames []string
+	//pg_sequences table is available from PG 10 and consist info of normal sequences and sequences generated by identity columns
 	query := fmt.Sprintf(`SELECT sequencename FROM pg_sequences where schemaname = '%s';`, schemaName)
 	rows, err := pg.db.Query(query)
 	if err != nil {
-		return nil, fmt.Errorf("error in querying(%q) source database for sequence names: %v", query, err)
+		if strings.Contains(err.Error(), "does not exist") {
+			//For PG version before 10 as identity columns are also introduced in PG 10 so using information_schema.sequences should be fine
+			query = fmt.Sprintf(`SELECT sequence_name FROM information_schema.sequences where sequence_schema = '%s';`, schemaName)
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply yb-voyager/src/srcdb/postgres.go:441]: Yes it is available in all PG versions, but we can't use this only here as it doesn't have the information about the identity columns which are introduced in PG10 and available in `pg_sequences` table.

File: yb-voyager/src/srcdb/postgres.go [hunk 3]
```diff
@@ -432,10 +432,20 @@ func (pg *PostgreSQL) GetAllSequences() []string {
 // GetAllSequencesRaw returns all the sequence names in the database for the schema
 func (pg *PostgreSQL) GetAllSequencesRaw(schemaName string) ([]string, error) {
 	var sequenceNames []string
+	//pg_sequences table is available from PG 10 and consist info of normal sequences and sequences generated by identity columns
 	query := fmt.Sprintf(`SELECT sequencename FROM pg_sequences where schemaname = '%s';`, schemaName)
 	rows, err := pg.db.Query(query)
 	if err != nil {
-		return nil, fmt.Errorf("error in querying(%q) source database for sequence names: %v", query, err)
+		if strings.Contains(err.Error(), "does not exist") {
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply yb-voyager/src/srcdb/postgres.go:439]: there are only few occurrences of this and that too in different packages so not of much of a value to have it.

**Pull request merged**